hcl/hcl_test.go

```diff
@@ -1100,14 +1291,23 @@ func testParser(t *testing.T, tc testcase) {
 		for _, inputConfigFile := range tc.input {
 			filename := inputConfigFile.filename
 			if filename == "" {
-				filename = config.DefaultFilename
+				panic("expect a filename in the input config")
 			}
-			cfgfile := test.WriteFile(t, configsDir, filename, inputConfigFile.body)
-			addFilenameToErrorsFileRanges(tc.want.errs, cfgfile)
+			test.WriteFile(t, configsDir, filename, inputConfigFile.body)
 		}
+		fixupFiledirOnErrorsFileRanges(configsDir, tc.want.errs)
 		got, err := hcl.ParseDir(configsDir)
 		errtest.AssertErrorList(t, err, tc.want.errs)
 
+		var gotErrs *errors.List
+		if errors.As(err, &gotErrs) {
+			if len(gotErrs.Errors()) != len(tc.want.errs) {
+				t.Logf("got errors: %s", gotErrs.Detailed())
```

Comment:
Logging wanted errors would help debugging when this fails (instead of just got).